src/iceberg/expressions/and.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <format>
+#include <memory>
+#include <string>
+
+#include "iceberg/expression.h"
+
+namespace iceberg {
+
+/// \brief An Expression that represents a logical AND operation between two expressions.
+///
+/// This expression evaluates to true if and only if both of its child expressions
+/// evaluate to true.
+class ICEBERG_EXPORT And : public Expression {
```

Comment:
Can we put all these expr impls into a single file? I'm afraid that we'll end up with too many files merely for exprs.